sdk/core/azure-core/src/main/java/com/azure/core/http/rest/RestProxy.java

```diff
@@ -124,17 +245,44 @@ public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline)
     /**
      * Create a proxy implementation of the provided Swagger interface.
      *
-     * @param swaggerInterface the Swagger interface to provide a proxy implementation for
-     * @param httpPipeline the HttpPipelinePolicy and HttpClient pipline that will be used to send Http requests
+     * @param swaggerInterface The class that represents a Swagger interfaces that will be turned into a proxy.
+     * @param httpPipeline the HttpPipelinePolicy and HttpClient pipeline that will be used to send Http requests
      * @param serializer the serializer that will be used to convert POJOs to and from request and response bodies
      * @param <A> the type of the Swagger interface.
      * @return a proxy implementation of the provided Swagger interface
      */
     @SuppressWarnings("unchecked")
     public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline, SerializerAdapter serializer) {
-        final SwaggerInterfaceParser interfaceParser = SwaggerInterfaceParser.getInstance(swaggerInterface);
-        final RestProxy restProxy = new RestProxy(httpPipeline, serializer, interfaceParser);
+        final RestProxy restProxy = new RestProxy(httpPipeline, serializer, swaggerInterface);
         return (A) Proxy.newProxyInstance(swaggerInterface.getClassLoader(), new Class<?>[]{swaggerInterface},
             restProxy);
     }
+
+    @FunctionalInterface
+    private interface AsyncExperimentalConstructor {
```

Comment:
instead of extending `AsyncRestProxy` perhaps it would be good to extract `updateRequest` logic into separate type and allow to inject customization via AsyncProxy ctor.

or perhaps AsyncRestProxy should become SPI ?

Reply:
I'll play with that, the design isn't final and I am looking for cleaner ways to do it